cli/types/cypress.d.ts

```diff
@@ -462,30 +465,98 @@ declare namespace Cypress {
      */
     log(options: Partial<LogConfig>): Log
 
-    /**
-     * @see https://on.cypress.io/api/commands
-     */
     Commands: {
+      /**
+       * Add a custom command
+       * @see https://on.cypress.io/api/commands
+       */
       add<T extends keyof Chainable>(name: T, fn: CommandFn<T>): void
+
+      /**
+       * Add a custom parent command
+       * @see https://on.cypress.io/api/commands#Parent-Commands
+       */
       add<T extends keyof Chainable>(name: T, options: CommandOptions & {prevSubject: false}, fn: CommandFn<T>): void
+
+      /**
+       * Add a custom child command
+       * @see https://on.cypress.io/api/commands#Child-Commands
+       */
       add<T extends keyof Chainable, S = any>(name: T, options: CommandOptions & {prevSubject: true}, fn: CommandFnWithSubject<T, S>): void
+
+      /**
+       * Add a custom command that validates the prevSubject
```

Comment:
I thought this added a custom command that could be used as a parent or a chained child command?

Reply:
Hm. Looking at the docs more closely, it looks like this can be used to add *either* a child or a dual command - in both cases you pass in prevSubject. It's dual if it includes `optional`, and a child if it does not. Updating comment. :+1:

Reply:
The more I look at it, the more I think this type is actually just malformed.  This case covers the following:

```
.add('foo', { prevSubject: ['optional'] }, ...) // A dual command
.add('foo', { prevSubject: 'window' }, ...) // A child command that only allows one type of subject
```

vs. the one below it, which covers:

```
.add('foo', { prevSubject: ['optional'] }, ...) // A dual command - the same case as above.
.add('foo', { prevSubject: ['element', 'document'] }, ...) // A child command
.add('foo', { prevSubject: ['element', 'optional'] }, ...) // A dual command that can be either a parent or a child taking an element.
```

The `| ['optional']` in this line isn't necessary; the case below already covers it.

Reply:
I reverted the change to the types; it started getting a bit larger than I expected. But still updated the comment. :+1: